From ae9a40f97815aa405fe3f1ceb6b5a330a150700c Mon Sep 17 00:00:00 2001
From: "Hans J. Johnson" <hans-johnson@uiowa.edu>
Date: Sat, 28 Mar 2026 09:05:35 -0500
Subject: [PATCH] COMP: Detect both QtTesting and qttesting CMake target names

The upstream QtTesting target name changed from "QtTesting" (pre-2024)
to lowercase "qttesting" in newer revisions. Add an if(TARGET qttesting)
check in Libs/QtTesting, Libs/Widgets, and Libs/Visualization/VTK/Widgets
so both the old Qt5 pin and the new Qt6-compatible revision link correctly.

Changes suggested by AI coding tools
---
 Libs/QtTesting/CMakeLists.txt                 | 11 +++++++----
 Libs/Visualization/VTK/Widgets/CMakeLists.txt |  8 +++++++-
 Libs/Widgets/CMakeLists.txt                   |  8 +++++++-
 3 files changed, 21 insertions(+), 6 deletions(-)

diff --git a/Libs/QtTesting/CMakeLists.txt b/Libs/QtTesting/CMakeLists.txt
index 0dc0ed5ba0..431bcae284 100644
--- a/Libs/QtTesting/CMakeLists.txt
+++ b/Libs/QtTesting/CMakeLists.txt
@@ -4,7 +4,13 @@ project(CTKQtTesting)
 # 3rd party dependencies
 #
 find_package(QtTesting REQUIRED)
-set(QtTesting_LIBRARIES QtTesting)
+# The upstream target name changed from "QtTesting" (pre-2024) to "qttesting"
+# (lowercase) in newer revisions. Use whichever target the package provides.
+if(TARGET qttesting)
+  set(QtTesting_LIBRARIES qttesting)
+else()
+  set(QtTesting_LIBRARIES QtTesting)
+endif()
 
 #
 # See CTK/CMake/ctkMacroBuildLib.cmake for details
@@ -135,9 +141,6 @@ set(KIT_resources
   Resources/ctkQtTesting.qrc
 )
 
-# Set QtTesting_LIBRARIES variable
-set(QtTesting_LIBRARIES QtTesting)
-
 # Target libraries - See CMake/ctkFunctionGetTargetLibraries.cmake
 # The following macro will read the target libraries from the file 'target_libraries.cmake'
 ctkFunctionGetTargetLibraries(KIT_target_libraries)
diff --git a/Libs/Visualization/VTK/Widgets/CMakeLists.txt b/Libs/Visualization/VTK/Widgets/CMakeLists.txt
index c66b32e203..bdfc0b9e9a 100644
--- a/Libs/Visualization/VTK/Widgets/CMakeLists.txt
+++ b/Libs/Visualization/VTK/Widgets/CMakeLists.txt
@@ -56,6 +56,12 @@ set(KIT_export_directive "CTK_VISUALIZATION_VTK_WIDGETS_EXPORT")
 #
 if(CTK_USE_QTTESTING)
   find_package(QtTesting REQUIRED)
+  # Target name changed from "QtTesting" (old) to "qttesting" (new upstream)
+  if(TARGET qttesting)
+    set(QtTesting_LINK_TARGET qttesting)
+  else()
+    set(QtTesting_LINK_TARGET QtTesting)
+  endif()
   include_directories(
     ${QtTesting_INCLUDE_DIRS}
     ${CMAKE_BINARY_DIR} # For ctkConfig.h
@@ -211,7 +217,7 @@ if(CTK_USE_QTTESTING)
     ctkVTKRenderViewEventPlayer.h
     ctkVTKRenderViewEventTranslator.h
 )
-  list(APPEND KIT_target_libraries QtTesting)
+  list(APPEND KIT_target_libraries ${QtTesting_LINK_TARGET})
 endif()
 
 # Prefer QVTKOpenGLWidget to QVTKWidget when using Qt5
diff --git a/Libs/Widgets/CMakeLists.txt b/Libs/Widgets/CMakeLists.txt
index 591e2d583a..02c9a50d11 100644
--- a/Libs/Widgets/CMakeLists.txt
+++ b/Libs/Widgets/CMakeLists.txt
@@ -18,6 +18,12 @@ endif()
 #
 if(CTK_USE_QTTESTING)
   find_package(QtTesting REQUIRED)
+  # Target name changed from "QtTesting" (old) to "qttesting" (new upstream)
+  if(TARGET qttesting)
+    set(QtTesting_LINK_TARGET qttesting)
+  else()
+    set(QtTesting_LINK_TARGET QtTesting)
+  endif()
   include_directories(
     ${QtTesting_INCLUDE_DIRS}
     ${CMAKE_BINARY_DIR} # For ctkConfig.h
@@ -311,7 +317,7 @@ if(CTK_USE_QTTESTING)
     ctkTreeComboBoxEventTranslator.cpp
     ctkTreeComboBoxEventTranslator.h
     )
-  list(APPEND KIT_target_libraries QtTesting)
+  list(APPEND KIT_target_libraries ${QtTesting_LINK_TARGET})
 endif()
 
 ctkMacroBuildLib(
